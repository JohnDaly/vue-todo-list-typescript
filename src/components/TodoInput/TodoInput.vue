<template>
    <b-field>
        <b-input v-model='newItem' expanded @keyup.native.enter='addItem' />
        <p class='control'>
            <button class='button is-primary' @click='addItem'>
                Add
            </button>
        </p>
    </b-field>
</template>

<script lang="ts" src="./TodoInput.ts"></script>

<style scoped lang="scss"></style>