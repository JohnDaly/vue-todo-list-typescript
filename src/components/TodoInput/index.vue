<template src='./template.html' />
<script lang="ts" src="./script.ts" />
<style scoped lang="scss" />