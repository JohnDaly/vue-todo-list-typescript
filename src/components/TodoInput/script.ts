// External Dependencies
import { Component, Vue, Emit } from 'vue-property-decorator';

@Component
export default class TodoInput extends Vue {
    // Initial Data
    newItem = '';

    @Emit('addItem')
    addItem() {
        const valueToEmit = this.newItem;
        this.newItem = '';
        return valueToEmit;
    }
}
