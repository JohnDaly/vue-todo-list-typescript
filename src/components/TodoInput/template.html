<b-field>
    <!-- The input field for adding a new todo to the list -->
    <b-input
        v-model='newItem'
        expanded
        @keyup.native.enter='addItem'
    />

    <!-- A button that will add the todo to the list -->
    <p class='control'>
        <button class='button is-primary' @click='addItem'>
            Add
        </button>
    </p>
</b-field>