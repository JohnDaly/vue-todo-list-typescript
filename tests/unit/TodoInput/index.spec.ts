// External Dependencies
import { shallowMount } from '@vue/test-utils';

// Internal Dependencies
import TodoInput from '@/components/TodoInput/index.vue';

describe('TodoInput.vue', () => {
    it('should render properly', () => {
        const wrapper = shallowMount(TodoInput, {});
        expect(wrapper).toMatchSnapshot();
    });
});
