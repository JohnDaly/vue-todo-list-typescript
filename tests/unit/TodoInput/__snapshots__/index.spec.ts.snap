// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput.vue should render properly 1`] = `
<b-field-stub addons="true">
  <b-input-stub expanded="true" usehtml5validation="true" value="" type="text" hascounter="true" customclass=""></b-input-stub>
  <p class="control"><button disabled="disabled" class="button is-primary">
      Add
    </button></p>
</b-field-stub>
`;
